tests/demos/test_demos_run.py

```diff
@@ -70,67 +130,38 @@ def py_file(rst_file, tmpdir, monkeypatch):
 
     # Get the name of the python file that pylit will make
     name = splitext(basename(rst_file))[0]
-    output = str(tmpdir.join(name))
+    py_file = str(tmpdir.join(name))
     # Convert rst demo to runnable python file
-    subprocess.check_call(["pylit", rst_file, output])
-    return output
+    subprocess.check_call(["pylit", rst_file, py_file])
+    return Path(py_file)
 
 
-@pytest.mark.skipcomplex  # Will need to add a seperate case for a complex demo.
-def test_demo_runs(py_file, env):
-    # Add pytest skips for missing imports or packages
-    if basename(py_file) in ("stokes.py", "rayleigh-benard.py", "saddle_point_systems.py", "navier_stokes.py", "netgen_mesh.py"):
-        if "mumps" not in get_external_packages():
-            pytest.skip("MUMPS not installed with PETSc")
+def _exec_file(py_file):
+    # To execute a file we import it. We therefore need to modify sys.path so the
+    # tempdir can be found.
+    sys.path.insert(0, str(py_file.parent))
+    importlib.import_module(py_file.with_suffix("").name)
+    sys.path.pop(0)  # cleanup
```

Comment:
To me this is way more elegant than `subprocess.check_call` (which happened to be breaking)

Reply:
This is :sparkles:magic:sparkles: